### PR TITLE
A fix to correct unique disconnect tracking

### PR DESCRIPTION
**Pull request by jessefriedman**
Previously connected sites (updated sites) will start with a value of -1 for `jetpack_unique_connection[disconnect]`.  This way we can differentiate between previously connected sites and newly connected sites while keeping the disconnect count accurate

Unique disconnects will only be tracked if the site was connected for the first time after 3.6

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -2336,17 +2336,21 @@ public static function disconnect( $update_activated_state = true ) {
 
 		$jetpack_unique_connection = Jetpack_Options::get_option( 'unique_connection' );
 		// Check then record unique disconnection if site has never been disconnected previously
-		if ( $jetpack_unique_connection['disconnected'] < 1 ) {
-
-			//track unique disconnect
-			$jetpack = Jetpack::init();
+		if ( $jetpack_unique_connection['disconnected'] == -1 ) {
```
Comment (dereksmart) [inline class.jetpack.php]: yoda plz

File: class.jetpack.php [hunk 2]
```diff
@@ -2336,17 +2336,21 @@ public static function disconnect( $update_activated_state = true ) {
...
+			$jetpack_unique_connection['disconnected'] = 1;
+		}
+		else {
+			if ( $jetpack_unique_connection['disconnected'] == 0 ) {
```
Comment (dereksmart) [inline class.jetpack.php]: yoda style, this should be :yoda:

Comment (jessefriedman) [inline class.jetpack.php]: fixed and tested

File: class.jetpack.php (hunk 1 above)
Comment (jessefriedman) [inline class.jetpack.php]: fixed and tested

File: class.jetpack.php [hunk 3]
```diff
@@ -378,10 +378,14 @@ function plugin_upgrade() {
 			}
 
 			//if Jetpack is connected check if jetpack_unique_connection exists and if not then set it
-			if ( ! get_option( 'jetpack_unique_connection' ) ) {
+			$jetpack_unique_connection = get_option( 'jetpack_unique_connection' );
+			if ( $jetpack_unique_connection && array_key_exists( 'version', $jetpack_unique_connection ) ) {
+				return;
```
Comment (dereksmart) [inline class.jetpack.php]: I don't think we should return here because there is more code that is supposed to run after this conditional

**Comment by dereksmart** [comment]
:+1: tested and works as expected

**Pull request merged**